libV2/schemaUtils.js

```diff
@@ -1092,8 +1101,51 @@ let QUERYPARAM = 'query',
    * @returns {Array} Examples for corresponding operation
    */
   generateExamples = (context, responseExamples, requestBodyExamples, responseBodySchema, isXMLExample) => {
-    const pmExamples = [];
+    const pmExamples = [],
+      responseExampleKeys = _.map(responseExamples, 'key'),
+      requestBodyExampleKeys = _.map(requestBodyExamples, 'key'),
+      matchedKeys = _.intersectionBy(responseExampleKeys, requestBodyExampleKeys, _.toLower),
+      usedRequestExamples = _.fill(Array(requestBodyExamples.length), false),
+      exampleKeyComparator = (example, key) => {
+        return _.toLower(example.key) === _.toLower(key);
+      };
+
+    // Do keys matching first and ignore any leftover req/res body for which matching is not found
+    if (matchedKeys.length) {
+      _.forEach(matchedKeys, (key) => {
+        const matchedRequestExamples = _.filter(requestBodyExamples, (example) => {
+            return exampleKeyComparator(example, key);
+          }),
+          responseExample = _.find(responseExamples, (example) => {
+            return exampleKeyComparator(example, key);
+          });
+
+        let requestExample = _.find(matchedRequestExamples, ['contentType', _.get(responseExample, 'contentType')]),
```

Comment:
@VShingala What's the reason for doing this? The request and response can be of different types. Why give preference to same content types?

Reply:
@webholik Yes, we only give preference to same content type if available. Reasoning is, It makes better example if both request and response body are in same content type.

Although, if same content type is not present we do pair different content types as well.